gazelle_haskell_modules/lang.go

```diff
@@ -213,10 +213,29 @@ func (*gazelleHaskellModulesLang) Fix(c *config.Config, f *rule.File) {
 				r.Delete()
 			}
 		}
+		fixModuleLists(r, ruleNameSet)
 	}
+
 	f.Sync()
 }
 
+// Leaves only those modules from r that are in ruleNameSet.
+// If a module is not in ruleNameSet, that means that there was no info associated with it,
+// hence it should be deleted.
+func fixModuleLists(r *rule.Rule, ruleNameSet map[string]bool) {
+	modules := r.AttrStrings("modules")
+	if modules != nil {
+		nonDeletedModules := make([]string, 0, len(modules))
+		for _, module := range modules {
+			// TODO: use labels instead of manually stripping away the ':' ?
+			if ruleNameSet[module[1:]] {
```

Comment:
In the latest version of the code perhaps something like this would do:
```diff
-       stripColon := func(module string) string { return module[1:] }
+       stripColon := func(module string) string {
+               if strings.HasPrefix(module, ":") {
+                       return module[1:]
+               } else {
+                       return module
+               }
+       }
```

Reply:
I tried a fix in 8e5ab77

Reply:
Actually, I think that right now (and before the patch too), if a user writes a fully qualified bazel target in `modules`, it will get deleted, regardless of whether the module exists or not, because `ruleNameSet` won't have the fully qualified module name.

So there probably needs to be something more complex here - either store full target names in `ruleNameSet`, or do a more complex check here.

I think the fully qualified names might be a better solution, as this string manipulation stuff feels very brittle both here and for `hidden_modules` :/

Reply:
https://github.com/tweag/gazelle_haskell_modules/commit/8e5ab7701e2811c3709608fbd8fcf93b896780e7 

Seems to have broken CI - need to investigate.

Reply:
> I think the fully qualified names might be a better solution, as this string manipulation stuff feels very brittle both here and for hidden_modules :/

A question that helps determining the severity of issues in general is: what the best workaround is?
In light of this question, you are correct that absolute labels in `modules` can confuse the fix command, but it is also easy for the user to make the labels relative instead, and relative labels are the most economical to use anyway. So it is unlikely to be a barrier for adoption.

Reply:
> Seems to have broken CI - need to investigate.

Mea culpa :see_no_evil: 
Fixed in efae1e7

Reply:
And still I was dropping non-relative labels. I tried another fix in e18e2de.